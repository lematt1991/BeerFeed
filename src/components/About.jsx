import React, {Component} from 'react';
import {Row, Jumbotron, Button} from 'react-bootstrap';
import QA from '../components/QA';
import LinkAccount from '../components/LinkAccount';

export default class About extends Component{
	constructor(){
		super()
		this.state = {
			buttonMsg : 'Help Improve Beer Feed',
			child : <QA/>,
			QA : true
		}
	}

	_handleClick(event){
		if(this.state.QA){
			this.setState({
				buttonMsg : 'FAQ',
				child : <LinkAccount/>,
				QA : false
			})
		}else{
			this.setState({
				buttonMsg : 'Help Improve Beer Feed',
				child : <QA/>,
				QA : true
			})
		}
	}

	render(){
		return(
			<div class="container">
				<Row>
				<Jumbotron>
				    <h1>Find better beer.</h1>
				    <p>
				    	Get a realtime feed of <i>only</i> the good beers that get 
				    	checked in to Untappd
				    </p>
				    <Button class="btn btn-primary" onClick={this._handleClick.bind(this)}>
				    	{this.state.buttonMsg}
				    </Button>	
				</Jumbotron>
				</Row>
				{this.state.child}
				<br/><br/>
			</div>
		)
	}
}