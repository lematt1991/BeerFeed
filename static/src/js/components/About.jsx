import React, {Component} from 'react';
import {Row, Jumbotron, Col, Button} from 'react-bootstrap';
import QA from 'beerfeed/components/QA';
import LinkAccount from 'beerfeed/components/LinkAccount';

const childID = {
	QA_ID : 0,
	LINK_ACCOUNT_ID : 1,
}

export default class About extends Component{

	constructor(){
		super()
		this.state = {
			buttonMsg : 'Help Improve Beer Feed',
			child : <QA/>,
			id : childID.QA_ID
		}
	}

	_handleClick(event){
		event.target.blur()
		if(this.state.id === childID.QA_ID){
			this.setState({
				buttonMsg : 'FAQ',
				child : <LinkAccount/>,
				id : childID.LINK_ACCOUNT_ID,
			})
		}else if(this.state.id === childID.LINK_ACCOUNT_ID){
			this.setState({
				buttonMsg : 'Help Improve Beer Feed',
				child : <QA/>,
				id : childID.QA_ID,
			})
		}else{
			console.log('ERROR!!!')
		}
	}

	render(){
		return(
			<div class="container">
				<Row>
				<Jumbotron>
				    <h1>Find better beer.</h1>
				    <p>
				    	Get a realtime feed of <i>only</i> the good beers that get 
				    	checked in to Untappd
				    </p>
				    <Button class="btn btn-primary" onClick={this._handleClick.bind(this)}>
				    	{this.state.buttonMsg}
				    </Button>	
				</Jumbotron>
				</Row>
				{this.state.child}
				<br/><br/>
			</div>
		)
	}
}