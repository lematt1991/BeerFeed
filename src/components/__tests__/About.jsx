import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import sinon from 'sinon'
import Adapter from 'enzyme-adapter-react-16'

import About from '../About'

Enzyme.configure({ adapter: new Adapter() });

describe('<About/>', () => {
	it('mounts', () => {
		var wrapper = shallow(<About/>)

		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('Has the QA component as a child by default', () => {
		var wrapper = shallow(<About/>)

		expect(wrapper.find('QA').exists()).toBeTruthy()
		expect(wrapper.find('LinkAccount').exists()).toBe(false)
	})

	it('Toggles between QA and LinkAccount', () => {
		var wrapper = shallow(<About/>)

		wrapper.find('Button').simulate('click')

		expect(wrapper.update().find('QA').exists()).toBe(false)
		expect(wrapper.update().find('LinkAccount').exists()).toBeTruthy()

		wrapper.find('Button').simulate('click')

		expect(wrapper.find('QA').exists()).toBeTruthy()
		expect(wrapper.find('LinkAccount').exists()).toBe(false)
	})
})