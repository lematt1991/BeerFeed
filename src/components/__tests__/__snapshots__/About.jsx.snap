// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About/> mounts 1`] = `
<div
  className="container"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Jumbotron
      bsClass="jumbotron"
      componentClass="div"
    >
      <h1>
        Find better beer.
      </h1>
      <p>
        Get a realtime feed of 
        <i>
          only
        </i>
         the good beers that get checked in to Untappd
      </p>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
      >
        Help Improve Beer Feed
      </Button>
    </Jumbotron>
  </Row>
  <QA />
  <br />
  <br />
</div>
`;
